PolyEngine/Engine/Src/Rendering/IRenderingDevice.hpp

```diff
@@ -64,14 +69,16 @@ namespace Poly
 	class ENGINE_DLLEXPORT ITextureDeviceProxy : public BaseObject<>
 	{
 	public:
-		virtual void SetContent(eTextureDataFormat format, const unsigned char* data) = 0;
-		virtual void SetSubContent(size_t width, size_t height, size_t offsetX, size_t offsetY, eTextureDataFormat format, const unsigned char* data) = 0;
+		virtual void SetContent(const unsigned char* data) = 0;
+		virtual void SetContentHDR(const float* data) = 0;
+		virtual void SetSubContent(size_t width, size_t height, size_t offsetX, size_t offsetY, const unsigned char* data) = 0;
+		virtual unsigned int GetResourceID() const = 0;
 	};
 
 	class ENGINE_DLLEXPORT ICubemapDeviceProxy : public BaseObject<>
 	{
 	public:
-		virtual void SetContent(const eCubemapSide side, const unsigned char* data) = 0;
+		virtual void SetContent(const eCubemapSide side, const float* data) = 0;
```

Comment:
Are cube maps usually represented as floats or do we need both methods?

Reply:
`float* data` stands for HDR image data. `unsigned char* data` stands for LDR image data. Will rename methods accordingly because depending on texture usage ASSERTE will trigger when wrong data is passed.